### PR TITLE
[72] Improve CSV export filename

### PR DESCRIPTION
**Pull request by caravinci**
As suggested in #72, this changes the filename for the exported CSV, to include details of the filters used to generate it.

### CODE REVIEW

File: billing-ui/src/pages/Report/Report.js [hunk 1]
```diff
@@ -123,12 +123,14 @@ const Report = ({ history }) => {
     const [entriesToDisplay, setEntriesToDisplay] = useState([]);
     const [reportSummary, setReportSummary] = useState([]);
     const [projects, setProjects] = useState(noProjects(report));
+    const [CSVFileName, setCSVFileName] = useState('');
```
Comment (caravinci) [inline billing-ui/src/pages/Report/Report.js:126]: Hold the filename for the currently generated report.

File: billing-ui/src/pages/Report/Report.js [hunk 2]
```diff
@@ -123,12 +123,14 @@ const Report = ({ history }) => {
...
     const [filters, setFilters] = useState({
         bucketSize: defaultBucketSize,
         fromDate: moment().subtract(1, 'months').startOf('day'),
         projects: [],
         toDate: moment().startOf('day'),
     });
+    const [canGenerateReport, setCanGenerateReport] = useState(filters.fromDate && filters.toDate);
```
Comment (caravinci) [inline billing-ui/src/pages/Report/Report.js:133]: Let's disable the "generate report" button, if any of the necessary filters is missing.

File: billing-ui/src/pages/Report/Report.js [hunk 3]
```diff
@@ -154,6 +156,53 @@ const Report = ({ history }) => {
         }
     };
 
+    const formatCSVFileName = () => setCSVFileName(() => {
```
Comment (caravinci) [inline billing-ui/src/pages/Report/Report.js:159]: Sets the date range to be added to the filename, based on the applied filters.

File: billing-ui/src/pages/Report/Report.js [hunk 4]
```diff
@@ -774,7 +830,7 @@ const Report = ({ history }) => {
                     ]}
                     data={entriesToDisplay}
                     exportCSV
-                    fileName="report.csv"
+                    fileName={`collaboratory_usage_billing_export_${CSVFileName}.csv`}
```
Comment (caravinci) [inline billing-ui/src/pages/Report/Report.js:833]: Applies the filename to the CSV

**Review by caravinci — COMMENTED** [review]
Some comments to help review

**Review by samrichca — APPROVED** [review]

**Pull request merged**